doc/classes/Shader.xml

```diff
@@ -55,6 +55,10 @@
 		<member name="code" type="String" setter="set_code" getter="get_code" default="&quot;&quot;">
 			Returns the shader's code as the user has written it, not the full generated code used internally.
 		</member>
+		<member name="custom_defines" type="String" setter="set_custom_defines" getter="get_custom_defines" default="&quot;&quot;">
+			Returns the shader's custom defines. Custom defines can be used in Godot to add GLSL preprocessor directives (e.g: extensions) required for the shader logic.
+			[b]Note:[/b] Custom defines are not validated by the Godot shader parser, so care should be taken when using them.
+		</member>
```

Comment:
Is this only for adding `ANDROID_ENABLED`, or is this intended to be used specifically by end users? It looks like a new features that is only partly related to this PR and would warrant separate design discussion.

If it's only for internal use of the engine, there's already an interface in place for setting defines from the rasterizer code (set `set_conditional` uses).

Reply:
This is intended to be used by end users since the external texture is meant to be used with a `ShaderMaterial` with a custom user defined shader. For example, to generate the video playback demo in the description, I pass the following to the custom defines to be applied to the user defined shader:

```
#ifdef ANDROID_ENABLED
#extension GL_OES_EGL_image_external : enable
#extension GL_OES_EGL_image_external_essl3 : enable
#else
#define samplerExternalOES sampler2D
#endif
```